### PR TITLE
Checkout: Add a cart icon to the Order Summary area

### PR DESCRIPTION
**Pull request by sixhours**
This PR adds a cart icon to the checkout area. This came up during our recent user research as something we could do to make it clearer the sidebar/Order Summary area is a cart and not a piece of navigation. 

**Before this PR:**

screen shot 2018-09-04 at 1 10 56 pm

**After this PR:**

screen_shot_2018-09-03_at_3 33 06_pm

I'd intended to add the cart to the left of "Order Summary", but it looks like the SectionHeader component accepts a string for the label text, no component objects.

Having the icon on the right kind of implies the cart icon is also an action that can be clicked, like the trash can icon.

We could fake it with a :before pseudo element or use CSS to reposition the icon to the left, but I don't see a pattern of using an icon in the SectionHeader elsewhere in Calypso, so maybe leaving it to the right is better for consistency.

What do you think @jancavan ?

### CODE REVIEW

File: client/my-sites/checkout/checkout/style.scss [hunk 1]
```diff
@@ -561,6 +561,10 @@
 	}
 }
 
+.cart__header .gridicons-cart {
+	transform: translate(.5px, .5px);
```
Comment (jancavan) [inline client/my-sites/checkout/checkout/style.scss]: Do you think we need `offset-adjust` here? It looks fine to me without it.

Comment (sixhours) [reply client/my-sites/checkout/checkout/style.scss]: Oh, I'm not sure...I was following the Gridicons guide in devdocs that said if you use icons at 18px they need offset adjust.

**Comment by sixhours** [comment]
@jancavan I think the color is currently black, since it's inherited from the heading.

File: client/my-sites/checkout/checkout/style.scss (hunk 1 above)
Comment (sixhours) [reply client/my-sites/checkout/checkout/style.scss]: Ohhh, I missed that. :D

**Pull request merged**